### PR TITLE
Problem: Support `amount` list on MsgSend & remove some `Todos`

### LINKED ISSUE #278: Problem: Support `amount` list on MsgSend

### PR DESCRIPTION
**Pull request by cdc-Hitesh**
Fixes #278

### CODE REVIEW

File: lib/src/transaction/msg/bank/msgsend.ts [hunk 1]
```diff
@@ -124,6 +123,5 @@ export const msgSend = function (config: InitConfigurations) {
 export type MsgSendOptions = {
     fromAddress: string;
     toAddress: string;
-    // Todo: It should be ICoin[]
-    amount: ICoin;
+    amount: ICoin[];
```
Comment (calvinaco) [inline lib/src/transaction/msg/bank/msgsend.ts:126]: Changing the signature will introduce breaking changes to existing users. How about introduces a new field `amounts: ICoin[]` on top of existing `amount: ICoin`?

Comment (cdc-Hitesh) [reply lib/src/transaction/msg/bank/msgsend.ts:126]: That's a good point though, but I worry about the possible confusions which may occur.
Here backward compatibility will come at a cost. 

Is indicating a breaking changelog enough for the new version users?
What do you say/

Comment (calvinaco) [reply lib/src/transaction/msg/bank/msgsend.ts:126]: We are following the https://rfc.zeromq.org/spec/42/#23-patch-requirements#26-evolution-of-public-contracts

All breaking changes to existing stable method should be introduced in a non-breaking fashion. I understand the confusion, and maybe we can introduces `MsgSendV2` instead?

Comment (calvinaco) [reply lib/src/transaction/msg/bank/msgsend.ts:126]: After all all messages and especially `MsgSend` is an important message type that involves funds movement and used by many parties. So it is important for us to introduce multi-asset without breaking it.

Comment (cdc-Hitesh) [reply lib/src/transaction/msg/bank/msgsend.ts:126]: This makes sense.
Thanks for the explanation.

Comment (cdc-Hitesh) [reply lib/src/transaction/msg/bank/msgsend.ts:126]: There are 2-3 other messages which will need to have a list supported on them.

I think we would need to create a V2 of all of such message types.

Comment (calvinaco) [reply lib/src/transaction/msg/bank/msgsend.ts:126]: Sure, thanks @cdc-Hitesh

Comment (cdc-Hitesh) [reply lib/src/transaction/msg/bank/msgsend.ts]: Resolved.

File: lib/src/transaction/msg/v2/bank/v2.msgsend.ts [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+import { v2 } from '../../ow.types';
+import { InitConfigurations, CroSDK } from '../../../../core/cro';
+import { AddressType, validateAddress } from '../../../../utils/address';
+import { CosmosMsg } from '../../cosmosMsg';
+import { COSMOS_MSG_TYPEURL } from '../../../common/constants/typeurl';
+import * as legacyAmino from '../../../../cosmos/amino';
+import { Network } from '../../../../network/network';
+
+export const msgSendV2 = function (config: InitConfigurations) {
+    return class MsgSend implements CosmosMsg {
```
Comment (calvinlauyh) [inline lib/src/transaction/msg/v2/bank/v2.msgsend.ts:14]:
```suggestion
    return class MsgSend implements CosmosMsg {
```

should we update here to `MsgSendV2` for better readability?

**Review by calvinlauyh — COMMENTED** [review]
Logic looks good to me! I notice most of the test cases and e2e test cases are covering single denom use cases. It would be good to cover multiple denoms.

We can create an new issue for this. Up to you @cdc-Hitesh

**Review by calvinaco — APPROVED** [review]

**Pull request merged**